extension/httpfs/src/cached_file_manager.cpp

```diff
@@ -45,13 +47,16 @@ std::string CachedFileManager::getCachedFilePath(const std::string& originalFile
     return common::stringFormat("{}/{}-{}", cacheDir, originalFileName, std::time(nullptr));
 }
 
-void CachedFileManager::downloadFile(const std::string& path, FileInfo* info) {
-    auto url = HTTPFileSystem::parseUrl(path);
-    httplib::Client cli(url.first);
-    httplib::Headers headers = {{}};
-    auto fileContent = cli.Get(url.second, headers)->body;
-    info->writeFile(reinterpret_cast<const uint8_t*>(fileContent.c_str()), fileContent.size(),
-        0 /* offset */);
+void CachedFileManager::downloadFile(HTTPFileInfo* fileToDownload, FileInfo* info) {
+    uint64_t numBytesRead;
+    uint64_t offsetToWrite = 0;
+    do {
+        numBytesRead = fileToDownload->readFile(downloadBuffer.get(), MAX_SEGMENT_SIZE);
+        info->writeFile(downloadBuffer.get(), numBytesRead, offsetToWrite);
+        offsetToWrite += numBytesRead;
+    } while (numBytesRead != 0);
```

Comment:
Let's do some benchmark here. I'm not sure about the performance implication here, but shouldn't a bit larger buffer size (but still multiple of `MAX_SEGMENT_SIZE`) will yield better throughput from httpfs/s3? Can u give a try on larger sizes?

Reply:
You are right, a relatively bigger segment size yields a better performance